### PR TITLE
Add minimal information

### PR DESCRIPTION
**Pull request by camappel**

### CODE REVIEW

**Comment by camappel** [comment]
Hi @acocac,

Just letting you know the minimal working version of the notebook is ready. It does the following:

- Fetch test data from Zenodo
- Download model from huggingface
- Predict on test set
- Train prebuilt model
- Download finetuned model from checkpoint on [huggingface](https://huggingface.co/camappel/deepforest-livestock/tree/main) ([PR](https://huggingface.co/weecology/deepforest-livestock/discussions/4))
- Evaluate finetuned model performance
- Visualize improvement

It could possible still do with more changes (please let me know if you think they're necessary) before merging:

- [ ] Load data as an intake catalogue
- [ ] Interactive visualisations (to show scores etc.)
- [ ] Transfer to new Binder (if you've implemented this already)
- [ ] Improved context section

Many thanks!

**Comment by acocac** [comment]
@camappel - I'm having a look at the minimal working version right now. I was on anual leave last week.

**Comment by acocac** [comment]
@camappel - I went through the notebook. Great work! I noted you indeed opened a PR to improve the version of the livestock model shared in Hugging Faces. Please find some suggestions below:

- Leave only the installation of DeepForest before loading the notebook libraries. To avoid conflicts because of future releases of DeepForest please declare the version when installing Deepforest e.g. `!pip -q install deepforest==1.3.3`. 
- Move PyTorch (`pytorch`, `torchvision`) and hugging face (`huggingface_hub`) dependencies to the `environment.yml` file. 
- The cell loading the baseline model fails `model.load_model(model_name="weecology/deepforest-livestock", revision="main")`. It looks the function is deprecated in the latest version of DeepForest.
   - A potential solution could be adding the `safetensors` in `environment.yml` and declare in the load libraries cell `import safetensors.torch`, then you should load the suggested safetensor file of the baseline model available in https://huggingface.co/weecology/deepforest-livestock/ using the following lines:

```
# download the baseline model
ckpt_path = hf_hub_download(
    repo_id="weecology/deepforest-livestock",  
    filename="model.safetensors"    
) 

# load state dictionary
state_dict = safetensors.torch.load_file(ckpt_path, device='cpu')

# load model
model = main.deepforest()
model.load_state_dict(state_dict)
```

- ~It seems the Binder run into out-of-memory (OOM) when evaluating the baseline and finetuned models. I'm wondered if you're also experiencing this.~
  - I just noted there's an easy-fix for the error. Jupyter notebooks [don't perform well with multiprocessing](https://stackoverflow.com/questions/68756034/pytorch-problem-my-jupyter-stuck-when-num-workers-0), so the `num_workers` should be changed to 0.
  - I suggest adding `model.config['workers'] = 0` after `model.config['gpus'] = '-1'`

Other improvements of the minimal working version:
- Ensure all dependencies are loaded at the beginning of notebook. Move the visualisation-related dependencies e.g. PIL, deepforest.visualize, matplotlib, os, pandas.
- Please indicate the dataset (cows) and codebase (DeepForest) version used in the Additional information section.

**Comment by acocac** [comment]
@camappel I wondered if you have completed the suggested changes. Happy to chat about them during the EDS monthly co-working session tomorrow.

**Comment by camappel** [comment]
Hi @acocac, thank so much for the comments and suggestions! I'm working on completing them at the moment and it's coming together very well. I also unfortunately won't be able to make today - will you be joining any of the coworking/collaboration calls in the near future?

Many thanks!

**Comment by acocac** [comment]
@camappel - thanks for the update. Please feel free to comment in the thread when the notebook is ready. Have a nice weekend 🌞

**Comment by camappel** [comment]
Hi @acocac, I've now made the suggested improvements, and it runs in Binder for me (eventually!) If you think it's ready, I'd be very happy to start the transfer process.

**Comment by acocac** [comment]
@camappel I successfully ran the notebook in the standard Binder, however, I noted the load operation of the pre-trained deepforest-livestock could take between 4-6 min. It'd be nice if you could add details on how models are loaded e.g. using HF safetensors and cross-reference the [webpage](https://huggingface.co/docs/safetensors/index) explaining why they're useful. Your notebook is the first in the EDS book gallery using such type of format so it might be beneficial for future contributors/readers to know why it's relevant.

Finally, while I appreciate being listed as a reviewer, please remove me if possible. My main role as Editors-in-chief is to ensure the notebook gets a minimum quality before starting the review process. Assigned reviewers will suggest further areas of improvement and we'll acknowledge their work accordingly.

**Comment by acocac** [comment]
@camappel - hope you're doing well. Please let me know if you'll implement further changes at this stage. Otherwise, I recommend the minimal working version of the notebook for the review process.

**Comment by camappel** [comment]
@acocac Sorry the delay getting back to - I have less free time now than in the summer! 

I've just pushed a couple of commits that address your comments. I'm also working on adding more detail to the training section (e.g. loss curves) but hopefully it's OK to push at a later stage.

Otherwise I'm very happy to start the review process!

Thanks

**Comment by acocac** [comment]
> @acocac Sorry the delay getting back to - I have less free time now than in the summer!
> 
> I've just pushed a couple of commits that address your comments. I'm also working on adding more detail to the training section (e.g. loss curves) but hopefully it's OK to push at a later stage.
> 
> Otherwise I'm very happy to start the review process!
> 
> Thanks

@camappel - thanks for addressing the comments. I'll start the review process soon. I'll tag you in a PRE-REVIEW issue to be opened in the core GitHub repository of EDS book, https://github.com/alan-turing-institute/environmental-ds-book.

**Comment by camappel** [comment]
Brilliant - thanks for all the support up to this point

**Comment by camappel** [comment]
@acocac the automatic commit [update](https://github.com/camappel/DeepForest/pull/1/commits/d792aa61807bb6de371f3affca377d86611e38cb) removed the dependancies and environment files, so I reverted to [Merge pull request #4 from eds-book-gallery/preparation](https://github.com/camappel/DeepForest/pull/1/commits/6e87de8efd907c411ecbe7bcb83b5368b7bd773f)

**Comment by acocac** [comment]
> @acocac the automatic commit [update](https://github.com/camappel/DeepForest/pull/1/commits/d792aa61807bb6de371f3affca377d86611e38cb) removed the dependancies and environment files, so I reverted to [Merge pull request #4 from eds-book-gallery/preparation](https://github.com/camappel/DeepForest/pull/1/commits/6e87de8efd907c411ecbe7bcb83b5368b7bd773f)

@camappel that's fine. Please open a PR to the review brach when the revised notebook is ready! thank you.

**Comment by camappel** [comment]
@acocac Do I need to merge into [main](https://github.com/camappel/DeepForest/tree/main) from [cam](https://github.com/camappel/DeepForest/tree/cam) so that we can merge into [camappel:main](https://github.com/camappel/DeepForest/tree/main) from [eds-book-gallery:postprint](https://github.com/eds-book-gallery/95199651-9e81-4cae-a3a7-66398a9a5f62/tree/postprint) as in #2 ?

**Comment by acocac** [comment]
@camappel please merge into main. Then I suggest to merge from [eds-book-gallery:main](https://github.com/eds-book-gallery/95199651-9e81-4cae-a3a7-66398a9a5f62/tree/main) into [camappel:main](https://github.com/camappel/DeepForest/tree/main) once the notebook is published.

**Comment by camappel** [comment]
Done!
